frontend/src/pages/Comparison.jsx

```diff
@@ -0,0 +1,166 @@
+import React, { useEffect } from 'react';
+import TestComparisonTable from '../components/comparisonTable/TestComparisonTable';
+import ComparisonCheckbox from '../components/comparisonTable/ComparisonCheckbox';
+import { useStateValue } from '../contexts/state';
+import MetadataTable from '../components/comparisonTable/MetadataTable';
+import { useParams } from 'react-router';
+import BreadcrumbNav from '../components/BreadcrumbNav';
+import ParentBuildComparison from '../components/comparisonTable/ParentBuildComparison';
+import ComparisonForm from './ComparisonForm';
+import Loading from '../components/Loading';
+import { ContainerGrid12, ContentGrid6 } from '../styles/baseComponents';
+import { TableContainer } from './Comparison.styles';
+
+const Build = () => {
+    const [{ loadingState }, dispatch] = useStateValue();
+
+    let { buildId, seriesId, buildId2, seriesId2 } = useParams();
+
+    useEffect(() => {
+        const fetchData = async () => {
+            dispatch({ type: 'setLoadingState', loadingState: true });
+            if (seriesId && buildId && seriesId2 && buildId2) {
+                try {
+                    const res = await fetch(
+                        `/data/series/${seriesId}/builds/${buildId}/metadata`,
+                        {}
+                    );
+                    let json = await res.json();
+
+                    const res2 = await fetch(
+                        `/data/series/${seriesId2}/builds/${buildId2}/metadata`,
+                        {}
+                    );
+                    let json2 = await res2.json();
+
+                    json2.metadata.forEach(meta => {
+                        const name = meta.metadata_name;
+                        let index = 0;
+                        let found = 0;
+                        let metadata_len = json.metadata.length;
+                        while (index < metadata_len && found === 0) {
+                            if (json.metadata[index].metadata_name === name) {
```

Comment:
```suggestion
                            if (json.metadata[parseInt(index)].metadata_name === name) {
```
This could fix the codacy issue